backfila-embedded/src/main/kotlin/app/cash/backfila/embedded/Backfila.kt

```diff
@@ -33,25 +33,110 @@ inline fun <reified Type : Backfill> Backfila.createDryRun(
   parameterData: Map<String, ByteString> = mapOf(),
   rangeStart: String? = null,
   rangeEnd: String? = null
+): BackfillRun<Type> {
+  return createDryRun(Type::class, parameters, parameterData, rangeStart, rangeEnd)
+}
+
+fun <Type : Backfill> Backfila.createDryRun(
+  backfill: KClass<Type>,
+  parameters: Any?,
+  parameterData: Map<String, ByteString>,
+  rangeStart: String?,
+  rangeEnd: String?
 ): BackfillRun<Type> {
   check(parameterData.isEmpty() || parameters == null) {
     "Only one of parameters and parameterData can be set"
   }
   val parameterBytes =
-    if (parameters != null) { parametersToBytes(parameters) } else { parameterData }
-  return createDryRun(Type::class, parameterBytes, rangeStart, rangeEnd)
+      if (parameters != null) {
+        parametersToBytes(parameters)
+      } else {
+        parameterData
+      }
+  return createDryRun(backfill, parameterBytes, rangeStart, rangeEnd)
 }
 
 inline fun <reified Type : Backfill> Backfila.createWetRun(
   parameters: Any? = null,
   parameterData: Map<String, ByteString> = mapOf(),
   rangeStart: String? = null,
   rangeEnd: String? = null
+): BackfillRun<Type> {
+  return createWetRun(Type::class, parameters, parameterData, rangeStart, rangeEnd)
+}
+
+fun <Type : Backfill> Backfila.createWetRun(
+  backfill: KClass<Type>,
+  parameters: Any?,
+  parameterData: Map<String, ByteString>,
+  rangeStart: String?,
+  rangeEnd: String?
 ): BackfillRun<Type> {
   check(parameterData.isEmpty() || parameters == null) {
     "Only one of parameters and parameterData can be set"
   }
   val parameterBytes =
-    if (parameters != null) { parametersToBytes(parameters) } else { parameterData }
-  return createWetRun(Type::class, parameterBytes, rangeStart, rangeEnd)
+      if (parameters != null) {
+        parametersToBytes(parameters)
+      } else {
+        parameterData
+      }
+  return createWetRun(backfill, parameterBytes, rangeStart, rangeEnd)
+}
+
+//**************************************************
+// Below are function signatures for easy java use.
+//**************************************************
+// Dry Run Signatures
+fun <Type : Backfill> Backfila.createDryRun(
+  backfill: Class<Type>,
+): BackfillRun<Type> {
+  return createDryRun(backfill.kotlin, null, mapOf(), null, null)
+}
+
+@JvmOverloads
+fun <Type : Backfill> Backfila.createDryRun(
```

Comment:
So this works, but I think it won't be as nice for Java callers as we thought. See https://stackoverflow.com/questions/28294509/accessing-kotlin-extension-functions-from-java

We'd end up having to call something like:

```java
@Inject Backfila backfila
...
BackfilaKt.createDryRun(backfila, MyBackfill.class)
```

which isn't great.